packages/wonder-blocks-dropdown/src/components/__tests__/single-select.test.tsx

```diff
@@ -1646,4 +1647,661 @@ describe("SingleSelect", () => {
+                // Arrange
+                const validate = jest.fn();
+                // Act
+                doRender(
+                    <ControlledSingleSelect
+                        validate={validate}
+                        selectedValue={"1"}
+                    />,
+                );
+                // Assert
+                expect(validate).toHaveBeenCalledExactlyOnceWith("1");
+            });
+            it("should be in an error state on mount if there is an invalid selected value", async () => {
+                // Arrange
+                // Act
+                doRender(
+                    <ControlledSingleSelect
+                        validate={() => "Error"}
+                        selectedValue={"1"}
+                    />,
+                );
+                // Assert
+                expect(await screen.findByRole("button")).toHaveAttribute(
+                    "aria-invalid",
+                    "true",
+                );
+            });
+            it("should not validate on mount if there is no selected value", () => {
+                // Arrange
+                // Act
+                const validate = jest.fn();
+                doRender(
+                    <ControlledSingleSelect
+                        validate={validate}
+                        selectedValue={undefined}
+                    />,
+                );
+
+                // Assert
+                expect(validate).not.toHaveBeenCalled();
+            });
+            it("should not be in an error state on mount if there is no selected value", async () => {
+                // Arrange
+                // Act
+                doRender(
+                    <ControlledSingleSelect
+                        validate={() => "Error"}
+                        selectedValue={undefined}
+                    />,
+                );
+                // Assert
+                expect(await screen.findByRole("button")).toHaveAttribute(
+                    "aria-invalid",
+                    "false",
+                );
+            });
+        });
+
+        describe("picking a value after there was a validation error", () => {
+            it("should still be in an error state before a value is picked", async () => {
+                // Arrange
+                const errorMessage = "Error message";
+                const userEvent = doRender(
+                    <ControlledSingleSelect
+                        validate={(value) =>
+                            value === "1" ? errorMessage : undefined
+                        }
+                        selectedValue={"1"}
+                    />,
+                );
+
+                // Act
+                await userEvent.click(await screen.findByRole("button")); // Open the dropdown
+
+                // Assert
+                expect(await screen.findByRole("button")).toHaveAttribute(
+                    "aria-invalid",
+                    "true",
+                );
```

Comment:
Good question - I was testing the error state before and after a value is picked when the select is already in an error state. The error state will stay if you open the dropdown, and it will go away only when a new valid value is picked. I updated the description block to `interactions after there is a validation error` instead to help clarify this. Let me know if that helps!

Reply:
That helps, thanks!